File: tools/other/mem0_helper.py
#!/usr/bin/env python3
import os, json, argparse
from pathlib import Path
from mem0 import MemoryClient

ENV_PATH = Path('/home/lazywork/.openclaw/workspace/.env.local')

def load_env():
    if ENV_PATH.exists():
        for line in ENV_PATH.read_text().splitlines():
            line=line.strip()
            if not line or line.startswith('#') or '=' not in line:
                continue
            k,v=line.split('=',1)
            os.environ.setdefault(k,v)

def client():
    load_env()
    return MemoryClient(api_key=os.environ['MEM0_API_KEY'])

def main():
    ap = argparse.ArgumentParser()
    sub = ap.add_subparsers(dest='cmd', required=True)
    a = sub.add_parser('add')
    a.add_argument('--user-id', required=True)
    a.add_argument('--message', required=True)
    a.add_argument('--assistant', default='')
    s = sub.add_parser('search')
    s.add_argument('--user-id', required=True)
    s.add_argument('--query', required=True)
    args = ap.parse_args()
    c = client()
    if args.cmd == 'add':
        messages = [{"role":"user","content":args.message}]
        if args.assistant:
            messages.append({"role":"assistant","content":args.assistant})
        print(json.dumps(c.add(messages, user_id=args.user_id), indent=2, default=str))
    elif args.cmd == 'search':
        filters = {"OR": [{"user_id": args.user_id}]}
        print(json.dumps(c.search(args.query, version='v2', filters=filters), indent=2, default=str))

if __name__ == '__main__':
    main()

File: tools/other/mem0_helper.js
#!/usr/bin/env node
const { MemoryClient } = require('mem0ai');

const apiKey = process.env.MEM0_API_KEY;
if (!apiKey) {
  console.error('MEM0_API_KEY is required');
  process.exit(1);
}

const client = new MemoryClient({ apiKey });

async function run() {
  const cmd = process.argv[2];
  const userId = process.env.MEM0_USER_ID || 'mr_o';

  if (cmd === 'add') {
    const text = process.argv.slice(3).join(' ');
    if (!text) throw new Error('Usage: mem0_helper.js add <text>');
    await client.add([
      { role: 'user', content: text },
      { role: 'assistant', content: 'Saved as critical memory.' }
    ], { user_id: userId, version: 'v2' });
    console.log('OK:add');
    return;
  }

  if (cmd === 'search') {
    const query = process.argv.slice(3).join(' ');
    if (!query) throw new Error('Usage: mem0_helper.js search <query>');
    const res = await client.search(query, { user_id: userId, version: 'v2' });
    console.log(JSON.stringify(res.slice(0, 5), null, 2));
    return;
  }

  if (cmd === 'seed-critical') {
    const critical = [
      '4-group watchlist mandatory: local tracked, api backend watchlist, rag v3 search, stocklist market-attractive.',
      'Morning brief must be sent via message() and include thought process per ticker.',
      'Unread alerts in queue must be sent immediately before other tasks.'
    ];
    for (const c of critical) {
      await client.add([{ role: 'user', content: c }], { user_id: userId, version: 'v2' });
    }
    console.log('OK:seed-critical');
    return;
  }

  throw new Error('Usage: mem0_helper.js <add|search|seed-critical> ...');
}

run().catch((e) => {
  console.error(e.message || String(e));
  process.exit(1);
});
